### PR TITLE
Envpak multiplatform upgrade - port floating point error handling to Linux

### PR DESCRIPTION
**Pull request by galanCA**
## Description

Updates ENVPAK.CPP to run on all platforms.

- Writes definitions for POSIX systems
- Suppresses function that is not declared
- Replaces windows function for C++ standandard.

## Author Progress Checklist:

- [x] Open draft pull request
    - [x] Make title clearly understandable in a standalone change log context
    - [x] Assign yourself the issue
    - [x] Add at least one label (enhancement, bug, or maintenance)
    - [ ] Link the issue(s) addressed by this PR (under "Development" in the sidebar menu) 
- [x] Make code changes (if you haven't already)
- [x] Self-review of code
    - [ ] My code follows the style guidelines of this project
    - [ ] I have added comments to my code, particularly in hard-to-understand areas
    - [ ] I have only committed the necessary changes for this fix or feature
    - [ ] I have made corresponding changes to the documentation
    - [ ] My changes generate no new warnings
    - [ ] I have ensured that my fix is effective or that my feature works as intended by:
        - [ ] exercising the code changes in the test framework, and/or
        - [ ] manually verifying the changes (as explained in the the pull request description above)
    - [ ] My changes pass all local tests
    - [ ] My changes successfully passes CI checks
    - [ ] Add any unit test for proof and documentation.
    - [ ] Merge in main branch and address resulting conflicts and/or test failures.
- [ ] Move pull request out of draft mode and assign reviewers
- [ ] Iterate with reviewers until all changes are approved
    - [ ] Make changes in response to reviewer comments
    - [ ] Merge in main branch and address resulting conflicts and/or test failures.
    - [ ] Re-request review in GitHub

## Reviewer Checklist:

 - [ ] Read the pull request description
 - [ ] Perform a code review on GitHub
 - [ ] Confirm all CI checks pass and there are no build warnings
 - [ ] Pull, build, and run automated tests locally
 - [ ] Perform manual tests of the fix or feature locally
 - [ ] Add any review comments, if applicable
 - [ ] Submit review in GitHub as either
     - [ ] Request changes, or
     - [ ] Approve
 - [ ] Iterate with author until all changes are approved

### CODE REVIEW

File: src/CNGLOB.H [hunk 1]
```diff
@@ -101,6 +101,51 @@ typedef HANDLE HINSTANCE;
 typedef HINSTANCE HMODULE;
 #endif
 
+#if defined(__linux__) || defined(__APPLE__)
```
Comment (tanaya-bigladder) [inline src/CNGLOB.H:104]: Could you elaborate a bit on how these are used? I didn't quite understand where they come from.

Comment (galanCA) [reply src/CNGLOB.H:104]: They are Windows definition and are primarily used for error reports. They can be found on lines [387](https://github.com/cse-sim/cse/blob/master/src/ENVPAK.CPP#L387-L393) and [423](https://github.com/cse-sim/cse/blob/master/src/ENVPAK.CPP#L423-L427).

Comment (tanaya-bigladder) [reply src/CNGLOB.H:104]: OK, this looks like something that will need a bit more research. I started [here](https://stackoverflow.com/questions/2219244/dealing-with-floating-point-exceptions). This is one of those places where it'll be important to separate what is platform-specific, vs what is compiler (or C++-library)-specific. I'm not well-versed in floating point exceptions, but I'll do some research as well.

Comment (tanaya-bigladder) [reply src/CNGLOB.H:104]: OK, this looks like something that will need a bit more research. I started [here](https://stackoverflow.com/questions/2219244/dealing-with-floating-point-exceptions). This is one of those places where it'll be important to separate what is platform-specific, vs what is compiler (or C++-library)-specific. I'm not well-versed in floating point exceptions, but I'll do some research as well.

**Comment by galanCA** [comment]
Should I remove [this ](https://github.com/cse-sim/cse/blob/f566feb8a5a81119e0c432dd43452a9ba5e45ba6/src/cseface.h#L25) comment?

**Comment by nealkruis** [comment]
@galanCA good question. The entire file you linked to is not used or maintained. Maybe it's best to leave it for now. I imagine we will redesign this when/if we want to separate out the CSE guts into a callable library.

**Comment by nealkruis** [comment]
@galanCA I think this one is waiting on a response from you. It also potentially relates to issue #343.

**Comment by galanCA** [comment]
@nealkruis would you like me to tackle issue #343 in this PR?

**Comment by nealkruis** [comment]
No, per my message in slack, I'm working on this already.

File: src/cnglob.h [hunk 2]
```diff
@@ -100,6 +100,51 @@ typedef HANDLE HINSTANCE;
...
+struct _exception
+{
+	int    type;   // exception type - see below
+	char* name;   // name of function where error occurred
+	double arg1;   // first argument to function
+	double arg2;   // second argument (if any) to function
+	double retval; // value to be returned by function
+};
+#endif // linux || APPLE
+
```
Comment (nealkruis) [inline src/cnglob.h:147]: Instead, look into removing the uses of these values when not compiling with MSVC.

**Review by nealkruis — CHANGES_REQUESTED** [review]
See comments re: definitions.

Also:

- merge master
- use the new native build configuration defintions

File: src/envpak.cpp [hunk 3]
```diff
@@ -462,9 +471,9 @@ int CheckFP( double v, const char* vTag)		// check for valid FP value
 // see CHECKFP macro (cnglob.h)
 // returns 0 if OK, else 1
 {	const char* t;
-	if (_isnan( v))
+	if (isnan( v))
```
Comment (tanaya-bigladder) [inline src/envpak.cpp:474]: shouldn't we use the std namespace here if we're using it below?

**Review by tanaya-bigladder — DISMISSED** [review]

File: src/envpak.cpp (hunk 3 above)
Comment (nealkruis) [reply src/envpak.cpp:473]: I was wondering the same thing. Is there an equivalent C function that is being used in this case? Probably can't hurt to use `std::`.

**Review by nealkruis — APPROVED** [review]

**Pull request merged**